addons/src/createNumberMask.js

```diff
@@ -27,6 +27,14 @@ export default function createNumberMask({
   const thousandsSeparatorSymbolLength = thousandsSeparatorSymbol && thousandsSeparatorSymbol.length || 0
 
   function numberMask(rawValue = emptyString) {
+    if(rawValue === '.' && allowDecimal) {
+      let mask = ['0', '.', /\d/]
+      if(prefix) {
+        mask.unshift(prefix)
```

Comment:
Hi Jackson, Thank you for such careful review.
Looks like I did miss quite a lot of scenarios if considering a few customizable configurations.

Let's see if we can come up a better solution to cover all possible scenarios.